packages/react-instantsearch-core/src/core/createConnector.js

```diff
@@ -162,7 +162,8 @@ export default function createConnector(connectorDesc) {
           this.context.ais.onSearchParameters(
             connectorDesc.getSearchParameters,
```

Comment:
Rather than providing the `this` as argument to the function we can bound the `this` to the function `getSearchParamters`. It avoids to pass the self on the parent function.